File: LeetCode/495-teemo-attacking/ex_495.py
class Solution(object):
    def findPoisonedDuration(self, timeSeries, duration):
        """
        :type timeSeries: List[int]
        :type duration: int
        :rtype: int
        """
        if not timeSeries:  return 0
        count = duration
        for i in range(1, len(timeSeries)):
            if timeSeries[i - 1] + duration <= timeSeries[i]:
                count += duration
            else:
                count += timeSeries[i] - timeSeries[i - 1]
        return count

File: LeetCode/495-teemo-attacking/ex_495.java
public class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        if (timeSeries.length == 0) return 0;
        int count = duration;
        for (int i = 1; i < timeSeries.length; i++) {
            if (timeSeries[i - 1] + duration <= timeSeries[i])
                count += duration;
            else
                count += timeSeries[i] - timeSeries[i - 1];
        }
        return count;
    }
}